src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+        {
+            foreach (var item in items)
+                list.Add(item);
+        }
+
+        public static void InsertRange(this IList list, object[] items)
+        {
+            for (int i = 0; i < items.Length; i++)
+                list.Insert(i, items[i]);
+        }
+
+        public static void InsertRange(this IList list, int listStartIndex, object[] items, int startIndex, int count)
+        {
+            var numToInsert = items.Length - startIndex;
+            if (count < numToInsert)
+                numToInsert = count;
+            for (int i = 0; i < numToInsert; i++)
+                list.Insert(listStartIndex + i, items[startIndex + i]);
+        }
+
+        public static void AddRange(this IList list, object[] items, int startIndex, int count)
+        {
+            var numToAdd = items.Length - startIndex;
+            if (count < numToAdd)
+                numToAdd = count;
+            for (int i = 0; i < numToAdd; i++)
+                list.Add(items[startIndex + i]);
+        }
+    }
+
+    public abstract class IListTest<T> : ICollectionTest<T>
+    {
+        protected bool ExpectedIsReadOnly { get; private set; }
+        protected bool ExpectedIsFixedSize { get; private set; }
+        protected bool SearchingThrowsFromInvalidValue { get; set; }
+        protected IListTest(bool isSynchronized, bool isReadOnly, bool isFixedSize) : base(isSynchronized)
+        {
+            ExpectedIsReadOnly = isReadOnly;
+            ExpectedIsFixedSize = isFixedSize;
+            SearchingThrowsFromInvalidValue = false; // This SHOULD be false but this is the current behavior
+        }
+
+        [Fact]
+        public void IsFixedSize()
+        {
+            Assert.Equal(ExpectedIsFixedSize, GetList(null).IsFixedSize);
+        }
+
+        [Fact]
+        public void IsReadOnly()
+        {
+            Assert.Equal(ExpectedIsReadOnly, GetList(null).IsReadOnly);
+        }
+
+        [Theory]
+        [InlineData(int.MinValue)]
+        [InlineData(-1)]
+        [InlineData(0)]
+        public void GetItemArgumentOutOfRange(int index)
+        {
+            IList list = GetList(null);
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () => list[index]);
+            CollectionAssert.Equal(Array.Empty<object>(), list);
+        }
+
+        [Fact]
+        public void GetItemArgumentOutOfRangeFilledCollection()
+        {
+            var items = GenerateItems(32);
+            IList list = GetList(items);
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () => list[list.Count]);
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void GetItemEmpty()
+        {
+            var items = new object[0];
+            var list = GetList(items);
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+            {
+                CollectionAssert.Equal(items, list);
```

Comment:
It asserts that the list and items are equal, using both IEnumerable, and IList functions.